lib/codegen/meta-python/base/legalize.py

```diff
@@ -382,6 +401,115 @@
             a << bxor(x, y)
         ))
 
+# Expand bitrev
+# Adapted from Stack Overflow.
+# https://stackoverflow.com/questions/746171/most-efficient-algorithm-for-bit-reversal-from-msb-lsb-to-lsb-msb-in-c
+widen.legalize(
+        a << bitrev.i8(x),
+        Rtl(
+            a1 << band_imm(x, imm64(0xaa)),
+            a2 << ushr_imm(a1, 1),
+            a3 << band_imm(x, imm64(0x55)),
+            a4 << ishl_imm(a3, 1),
+            b << bor(a2, a4),
+            b1 << band_imm(b, imm64(0xcc)),
+            b2 << ushr_imm(b1, 2),
+            b3 << band_imm(b, imm64(0x33)),
+            b4 << ushr_imm(b3, 2),
+            c << bor(b2, b4),
+            c1 << band_imm(c, imm64(0xf0)),
+            c2 << ushr_imm(c1, 4),
+            c3 << band_imm(c, imm64(0x0f)),
+            c4 << ishl_imm(c3, 4),
+            a << bor(c2, c4),
+        ))
+
+widen.legalize(
+        a << bitrev.i16(x),
+        Rtl(
+            a1 << band_imm(x, imm64(0xaaaa)),
+            a2 << ushr_imm(a1, 1),
```

Comment:
Ah, mypy is still failing because the shift immediates `1` etc. also need to be wrapped in imm64.

Reply:
@sunfishcode Doh, I've updated it now with all of them wrapped.